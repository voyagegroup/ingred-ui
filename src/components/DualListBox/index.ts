export { default } from "./DualListBox";
export type {
  DualListBoxProps,
  DualListBoxItem,
  DualListBoxCandidateItem,
} from "./DualListBox";
