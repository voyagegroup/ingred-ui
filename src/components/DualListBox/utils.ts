import {
  CandidateItem,
  DualListBoxCandidateItem,
  DualListBoxItem,
} from "./DualListBox";

export const getCandidateItems = (
  candidateItems: DualListBoxCandidateItem[],
  selectedItems: DualListBoxItem[],
) => {
  const result: CandidateItem[] = [];
  candidateItems.forEach((candidateItem) => {
    if (candidateItem.items) {
      result.push({
        ...candidateItem,
        items: getCandidateItems(candidateItem.items, selectedItems),
      });
      return;
    }

    const targetItem = selectedItems.find(
      (selectedItem) => selectedItem.id === candidateItem.id,
    );
    if (targetItem) {
      result.push({
        ...candidateItem,
        selected: true,
      });
    } else {
      result.push(candidateItem);
    }
  });

  return result;
};
