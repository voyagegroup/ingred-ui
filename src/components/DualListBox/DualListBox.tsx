import * as React from "react";
import * as Styled from "./styled";
import { CandidateRenderer } from "./internal/CandidateRenderer";
import { SelectedList } from "./internal/SelectedList";
import Divider from "../Divider/Divider";
import { useTheme } from "../../themes";
import { getCandidateItems } from "./utils";
import { useLocaleProps } from "../../hooks/useLocaleProps";

export type DualListBoxItem = {
  id: string;
  content: React.ReactNode;
};

export type DualListBoxCandidateItem = DualListBoxItem & {
  items?: DualListBoxCandidateItem[];
};

export type DualListBoxProps = {
  candidateItems: DualListBoxCandidateItem[];
  selectedItems: DualListBoxItem[];
  onAdd?: (item: DualListBoxItem) => void;
  onRemove?: (item: DualListBoxItem) => void;
};

/**
 * @memo 内部で選択候補の状態を保持するための型
 */
export type CandidateItem = DualListBoxCandidateItem & {
  selected?: boolean;
};

const DualListBox = React.forwardRef<HTMLDivElement, DualListBoxProps>(
  function DualListBox(inProps) {
    const {
      candidateItems: candidateItemsProp,
      selectedItems,
      selectedItemTitle,
      onAdd,
      onRemove,
    } = useLocaleProps({ props: inProps, name: "DualListBox" });

    const theme = useTheme();

    const candidateItems: CandidateItem[] = React.useMemo(
      () => getCandidateItems(candidateItemsProp, selectedItems),
      [candidateItemsProp, selectedItems],
    );

    const handleAdd = (item: DualListBoxItem) => {
      if (onAdd) {
        onAdd(item);
      }
    };

    const handleRemove = (item: DualListBoxItem) => {
      if (onRemove) {
        onRemove(item);
      }
    };

    return (
      <Styled.Container>
        <CandidateRenderer
          items={candidateItems}
          onAdd={handleAdd}
          onRemove={onRemove}
        />
        <Divider color={theme.palette.divider} orientation="vertical" />
        <SelectedList
          items={selectedItems}
          selectedItemTitle={selectedItemTitle}
          onRemove={handleRemove}
        />
      </Styled.Container>
    );
  },
);

export default DualListBox;
