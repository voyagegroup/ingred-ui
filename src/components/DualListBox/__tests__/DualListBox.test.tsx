import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import DualListBox from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("DualListBox component testing", () => {
  afterEach(cleanup);

  test("DualListBox", () => {
    const { asFragment } = renderWithThemeProvider(
      <DualListBox candidateItems={[]} selectedItems={[]} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DualListBox candidateItems", () => {
    const { asFragment } = renderWithThemeProvider(
      <DualListBox
        candidateItems={[
          {
            id: "1",
            label: "foo",
          },
          {
            id: "2",
            label: "bar",
          },
          {
            id: "3",
            label: "hoge",
          },
          {
            id: "4",
            label: "fuga",
          },
        ]}
        selectedItems={[]}
        onAdd={jest.fn()}
        onRemove={jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DualListBox selectedItems", () => {
    const { asFragment } = renderWithThemeProvider(
      <DualListBox
        candidateItems={[]}
        selectedItems={[
          {
            id: "1",
          },
          {
            id: "2",
          },
          {
            id: "3",
          },
          {
            id: "4",
          },
        ]}
        onAdd={jest.fn()}
        onRemove={jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DualListBox candidateItems and selectedItems", () => {
    const { asFragment } = renderWithThemeProvider(
      <DualListBox
        candidateItems={[
          {
            id: "1",
            label: "foo",
          },
          {
            id: "2",
            label: "bar",
          },
          {
            id: "3",
            label: "hoge",
          },
          {
            id: "4",
            label: "fuga",
          },
        ]}
        selectedItems={[
          {
            id: "1",
          },
          {
            id: "2",
          },
          {
            id: "3",
          },
          {
            id: "4",
          },
        ]}
        onAdd={jest.fn()}
        onRemove={jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DualListBox candidateItems and selectedItems with inverse", () => {
    const { asFragment } = renderWithThemeProvider(
      <DualListBox
        candidateItems={[
          {
            id: "1",
            label: "foo",
          },
          {
            id: "2",
            label: "bar",
          },
          {
            id: "3",
            label: "hoge",
          },
          {
            id: "4",
            label: "fuga",
          },
        ]}
        selectedItems={[
          {
            id: "1",
            checked: true,
          },
          {
            id: "2",
            checked: true,
          },
          {
            id: "3",
            checked: true,
          },
          {
            id: "4",
            checked: true,
          },
        ]}
        onAdd={jest.fn()}
        onRemove={jest.fn()}
        onToggleChange={jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
