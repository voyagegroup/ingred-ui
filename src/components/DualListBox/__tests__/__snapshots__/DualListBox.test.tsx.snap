// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DualListBox component testing DualListBox 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jKPMRM"
  >
    <ul
      class="sc-jqUVSM bBFcLq"
    />
    <hr
      class="sc-fLlhyt cCPNfc"
      color="#B3BAC1"
      orientation="vertical"
    />
    <div
      class="sc-bjUoiL hkPZz"
    >
      <div
        class="sc-idiyUo gIEPRB"
      >
        <p
          class="sc-fEOsli fayFCh"
          color="#041C33"
          font-size="13px"
        >
          <span
            class="sc-fEOsli jPgBVF"
            color="#0B82F4"
            font-size="13px"
          >
            0 
          </span>
          selected
        </p>
      </div>
      <ul
        class="sc-dIouRR"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DualListBox component testing DualListBox candidateItems 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jKPMRM"
  >
    <ul
      class="sc-jqUVSM bBFcLq"
    >
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                foo
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                bar
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                hoge
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie hlbDuP dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                fuga
              </p>
            </span>
          </label>
        </div>
      </li>
    </ul>
    <hr
      class="sc-fLlhyt cCPNfc"
      color="#B3BAC1"
      orientation="vertical"
    />
    <div
      class="sc-bjUoiL hkPZz"
    >
      <div
        class="sc-idiyUo gIEPRB"
      >
        <p
          class="sc-fEOsli fayFCh"
          color="#041C33"
          font-size="13px"
        >
          <span
            class="sc-fEOsli jPgBVF"
            color="#0B82F4"
            font-size="13px"
          >
            0 
          </span>
          selected
        </p>
      </div>
      <ul
        class="sc-dIouRR"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DualListBox component testing DualListBox candidateItems and selectedItems 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jKPMRM"
  >
    <ul
      class="sc-jqUVSM bBFcLq"
    >
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                foo
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                bar
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                hoge
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie hlbDuP dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                fuga
              </p>
            </span>
          </label>
        </div>
      </li>
    </ul>
    <hr
      class="sc-fLlhyt cCPNfc"
      color="#B3BAC1"
      orientation="vertical"
    />
    <div
      class="sc-bjUoiL hkPZz"
    >
      <div
        class="sc-idiyUo gIEPRB"
      >
        <p
          class="sc-fEOsli fayFCh"
          color="#041C33"
          font-size="13px"
        >
          <span
            class="sc-fEOsli jPgBVF"
            color="#0B82F4"
            font-size="13px"
          >
            4 
          </span>
          selected
        </p>
      </div>
      <ul
        class="sc-dIouRR"
      >
        <li
          class="sc-hHLeRK dPqaSF"
        >
          foo
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK dPqaSF"
        >
          bar
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK dPqaSF"
        >
          hoge
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK jXxJVM"
        >
          fuga
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DualListBox component testing DualListBox candidateItems and selectedItems with inverse 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jKPMRM"
  >
    <ul
      class="sc-jqUVSM bBFcLq"
    >
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                foo
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                bar
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie kxcdly dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                hoge
              </p>
            </span>
          </label>
        </div>
      </li>
      <li
        class="sc-kDDrLX sc-iqcoie hlbDuP dcUJGp"
      >
        <div
          class="sc-dkzDqf hETaFj"
          display="flex"
        >
          <label
            class="sc-hAZoDl cnpnnO"
          >
            <input
              checked=""
              class="sc-ksZaOG etQTLn"
              readonly=""
              type="checkbox"
            />
            <span
              class="sc-fnykZs hZrbBW"
            >
              <p
                class="sc-fEOsli irbPqM"
                color="#041C33"
                font-size="14px"
              >
                fuga
              </p>
            </span>
          </label>
        </div>
      </li>
    </ul>
    <hr
      class="sc-fLlhyt cCPNfc"
      color="#B3BAC1"
      orientation="vertical"
    />
    <div
      class="sc-bjUoiL hkPZz"
    >
      <div
        class="sc-idiyUo gIEPRB"
      >
        <p
          class="sc-fEOsli fayFCh"
          color="#041C33"
          font-size="13px"
        >
          <span
            class="sc-fEOsli jPgBVF"
            color="#0B82F4"
            font-size="13px"
          >
            4 
          </span>
          selected
        </p>
      </div>
      <ul
        class="sc-dIouRR"
      >
        <li
          class="sc-hHLeRK dPqaSF"
        >
          foo
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK dPqaSF"
        >
          bar
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK dPqaSF"
        >
          hoge
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK jXxJVM"
        >
          fuga
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DualListBox component testing DualListBox selectedItems 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jKPMRM"
  >
    <ul
      class="sc-jqUVSM bBFcLq"
    />
    <hr
      class="sc-fLlhyt cCPNfc"
      color="#B3BAC1"
      orientation="vertical"
    />
    <div
      class="sc-bjUoiL hkPZz"
    >
      <div
        class="sc-idiyUo gIEPRB"
      >
        <p
          class="sc-fEOsli fayFCh"
          color="#041C33"
          font-size="13px"
        >
          <span
            class="sc-fEOsli jPgBVF"
            color="#0B82F4"
            font-size="13px"
          >
            4 
          </span>
          selected
        </p>
      </div>
      <ul
        class="sc-dIouRR"
      >
        <li
          class="sc-hHLeRK dPqaSF"
        >
          foo
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK dPqaSF"
        >
          bar
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK dPqaSF"
        >
          hoge
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="sc-hHLeRK jXxJVM"
        >
          fuga
          <div
            class="sc-kgflAQ hZAlxa"
          >
            <button
              class="sc-dmRaPn mhBDH"
              type="button"
            >
              <span
                class="sc-gsnTZi bkuzTp"
                size="12"
              >
                <svg
                  fill="none"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.45456 1.45449C2.25788 0.65117 3.5605 0.651339 4.36403 1.45487L8.0005 5.09135L11.6374 1.45446C12.4405 0.651345 13.7426 0.651345 14.5457 1.45446L14.5465 1.45522C15.3496 2.25833 15.3496 3.56044 14.5465 4.36356L10.9096 8.00044L14.5447 11.6356C15.3483 12.4391 15.3484 13.7417 14.5451 14.545C13.7418 15.3484 12.4392 15.3482 11.6356 14.5447L8.0005 10.9095L4.36427 14.5458C3.56116 15.3489 2.25905 15.3489 1.45593 14.5458L1.45518 14.545C0.652059 13.7419 0.652059 12.4398 1.45517 11.6367L5.09141 8.00044L1.45493 4.36397C0.6514 3.56043 0.651231 2.25782 1.45456 1.45449Z"
                    fill="#041C33"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
